inc/Symbol.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SYMBOL_H
+#define SYMBOL_H
+
+#include <iostream>
+
+namespace Calculator {
+
+/**
+ * @brief Represents a symbol in a mathematical expression. Symbols can be any single character from a-z.
+ * 
+ */
+template<typename T>
+class Symbol {
+public:
+    const char symbol;
+private:
+    bool is_bound;
+    T bound_value;
+public:
+    Symbol() = delete;
+
+    Symbol(char c) : symbol(tolower(c)), is_bound(false) {
+        if(!isalpha(c)) throw new std::domain_error("given character was not a letter");
+    }
+
+    bool hasValue() {
```

Comment:
move implementation to .cpp - alternatively simple getters and setters can be inline functions. e.g.
```cpp
inline bool hasValue() {return isBound_;}
```

note: technically oneliners like this are discouraged, but it is common to see in code.